docs/contributing/commit-discipline.md

```diff
@@ -86,87 +86,224 @@ check out GitHub's excellent [blog post](https://github.blog/2022-06-30-write-be
+- `blueslip: Add feature to time common operations.`
+- `gather_subscriptions: Fix exception handling bad input.`
+- `stream_settings: Fix save/discard widget on narrow screens.`
+
+#### Detailed example
+
+- **Good summary**: "gather_subscriptions: Fix exception handling bad input."
+- **Not so good alternatives**:
+  - "gather_subscriptions was broken": This doesn't explain how it was broken, and
+    doesn't follow the format guidelines for commit summaries.
+  - "Fix exception when given bad input": It's impossible to tell what part of the
+    codebase was changed.
+  - Not using the imperative:
+    - "gather_subscriptions: Fixing exception when given bad input."
+    - "gather_subscriptions: Fixed exception when given bad input."
+
+### Commit description
+
+The body of the commit message should explain why and how the change
+was made. Like a good code comment, it should provide context and
+motivation that will help both a reviewer now, and a developer looking
+at your changes a year later, understand the motivation behind your
+decisions.
+
+Many decisions may be documented in multiple places (for example, both
+in a commit message and a code comment). The general rules of thumb are:
+
+- If the information will be helpful long-term for someone trying to
+  read and understand the code base, include it in a code comment.
+- If the information will be helpful long-term for someone trying to
+  investigate why these changes are a complete and correct
+  implementation of the commit's idea, include it in the commit
+  message.
+- If the information is helpful for reviewing your work (for example,
+  an alternative approach that you rejected or are considering,
+  something you noticed that seemed weird, or an error you aren't sure
+  you resolved correctly), include it in the PR description /
+  discussion.
+
+As an example, if you have a question that you expect to be resolved
+during the review process, put it in a PR comment attached to a
+relevant part of the changes. When the question is resolved, remember
+to update code comments and/or the commit description to document the
+reasoning behind the decisions.
+
+There are some cases when the best approach is improving the code or commit
+structure, not writing up details in a comment or a commit message. For example:
+
+- If the information is the description of a calculation or function,
+  consider the abstractions you're using. Often, a better name for a
+  variable or function is a better path to readable code than writing
+  a prose explanation.
+- If the information describes an additional change that you made while working
+  on the commit, consider whether it is separable from the rest of the project.
+  If it is, it should probably be moved to its own commit, with its own commit
+  message explaining it. It is far easier to review and integrate a series of
+  several well-written commits than it is to review those same changes in a
+  single commit.
```

Comment:
Felt like a lot of "it" in this bullet point, so revising the last sentence to be ...

> Reviewing and integrating a series of several well-written commits is far easier than reviewing those same changes in a single commit.